### PR TITLE
Add Drupal 9 to start states

### PR DESCRIPTION
**Pull request by EdwardAngert**
Closes: #DOC-311

## Summary

**[Creating Sites](https://pantheon.io/docs/create-sites)** - Customers can now create a new Drupal 9 site through the Dashboard! This Docs update is part of a larger effort to incorporate more Drupal 9 documentation.

## Effect

The following changes are already committed:

## Remaining Work

The following changes still need to be completed:

- [x] creating sites
- [x] start-state
- [x] create-custom-upstream
   - [x] we need a new drops-9 link / or [https://github.com/pantheon-upstreams/drupal-project](https://github.com/pantheon-upstreams/drupal-project)?
   - [x] def tech review requested here - this repo's default branch is `main` so I set the steps to fetch into `master` to keep consistent with other custom upstreams
- [x] ~drupal-9 (might need its own PR?)~ separate PR: #6467 

## Post Launch

**Do not remove** - To be completed by the docs team upon merge:

- [ ] Redirect `/docs/old-path/` => `/docs/new-path/` (if applicable)
- [ ] Include/exclude pages ^ respectively within docs search service provider (if applicable)
- [ ] For Heroes - add a props post to the [discussion board](https://discuss.pantheon.io/c/pantheon-platform/documentation/17).
- [ ] Remove from the [project board](https://github.com/pantheon-systems/documentation/projects/14)

### CODE REVIEW

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | [Live](https://pantheon.io/docs) | [pr-6389](https://pr-6389--pantheon-docs.my.pantheonfrontend.website/docs) | Difference |
| --- | --- | --- | --- |
| Performance Score | `52` | `67` | `15` |
| Best Practice Score | `79` | `79` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `96` | `96` | `0` |

Lighthouse Audit Reports:

- [`pr-6389` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/27237/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/27237/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)


[Preview Environment](https://pr-6389--pantheon-docs.my.pantheonfrontend.website/docs)

**Comment by EdwardAngert** [comment]
should we redirect the [D9](https://pantheon.io/docs/drupal-9) doc to the [creating sites](https://pantheon.io/docs/create-sites) doc since creating a D9 site is (or will be) just like creating any other site and add a link from there to the [integrated composer](https://pantheon.io/docs/integrated-composer) doc?

**Review by greg-1-anderson — APPROVED** [review]
This update is good and is correct as far as it goes. When working with an Integrated Composer upstream, there are additional instructions as described in the [IC docs](https://pantheon.io/docs/integrated-composer#how-to-add-dependencies-to-your-upstream) that are also relevant and should be referenced or copied in in some way.

**Comment by EdwardAngert** [comment]
> This update is good and is correct as far as it goes. When working with an Integrated Composer upstream, there are additional instructions as described in the [IC docs](https://pantheon.io/docs/integrated-composer#how-to-add-dependencies-to-your-upstream) that are also relevant and should be referenced or copied in in some way.

I agree re: the IC steps for the Custom Upstream, and was going to make it a separate PR, but I might be able to both borrow from the existing doc and link back to it. Does it seem like I captured the basics [in these steps](https://github.com/pantheon-systems/documentation/pull/6389/files#diff-eaacc4715497d1f58603b35892c08c346c06657b6ac3d107aeab462bfb804a0fR200-R216), @greg-1-anderson ?

**Comment by EdwardAngert** [comment]
while I was working on this one, I noticed that there was a similar recommendation made in #6197 by @dficker and @populist 
would what I have here be easily copied into the D8 tab and resolve both of these needs in one shot? 🤞

File: source/content/create-custom-upstream.md [hunk 1]
```diff
@@ -147,6 +147,14 @@ To avoid incompatibilities, you must track Pantheon's corresponding upstream rep
 
     </Tab>
 
+    <Tab title="Drupal 9" id="d91">
+
+    ```bash{promptUser: user}
+    git remote add pantheon-drupal-9 git://github.com/pantheon-upstreams/drupal-project.git
```
Comment (kyletaylored) [inline source/content/create-custom-upstream.md:153]: I believe the official D9 repo is https://github.com/pantheon-systems/drupal-project

File: source/content/create-custom-upstream.md [hunk 2]
```diff
@@ -180,6 +188,37 @@ To avoid incompatibilities, you must track Pantheon's corresponding upstream rep
...
+    1. Pull in the core files and commit them first:
+
+     ```bash{promptUser: user}
+     git checkout -b master
+     git fetch pantheon-drupal-9 master
+     git merge pantheon-drops-9/master
+     git push origin master
+     ```
+
+    1. Drupal 9 on Pantheon includes [Integrated Composer](/integrated-composer) to manage dependencies.
```
Comment (kyletaylored) [inline source/content/create-custom-upstream.md:202]: I would almost just send them to the Integrated Composer docs so this information isn't duplicated and have to be updated in multiple places later.

If not, just be explicit that they need to `cd upstream-config` before updating the packages in the upstream.

With this composer structure, we just need to be really clear that there are TWO `composer.json` files, and which one they should be updating.

**Comment by EdwardAngert** [comment]
after talking with Greg and @kyletaylored on slack about the `composer config version` step, I'm moving this back to "In Progress" while we work out whether that step is necessary/helpful or ultimately more confusing for users

File: source/content/create-custom-upstream.md (hunk 1 above)
Comment (greg-1-anderson) [reply source/content/create-custom-upstream.md:153]: `pantheon-upstreams/drupal-project` is correct.  `pantheon-systems/drupal-project` is the source repository that generates it.

File: source/content/create-custom-upstream.md (hunk 2 above)
Comment (EdwardAngert) [reply source/content/create-custom-upstream.md]: I added a `cd upstream-config` line, and don't `cd` back, but I'm not sure that's necessarily bad, considering the rest of the doc
moving the doc to tech review and we can change it if needed. I'll leave this conversation open for now

**Comment by EdwardAngert** [comment]
heads up: before we remove the partial that shows how to install D9 from the upstream, both this PR and #6467 should be merged
then we should edit the steps in [Manually Migrate D9](https://pantheon.io/docs/guides/drupal-9-migration/migrate-manual-d9#create-a-new-drupal-9-site) to say, ~"just create one from the Dashboard, friend"
doesn't _have_ to be in this order, I just don't want to delay this PR any further to adjust steps that work

**Review by joa-pan — APPROVED** [review]
LGTM

**Pull request merged**